packages/react/src/components/List/List.story.jsx

```diff
@@ -204,6 +213,65 @@ export const BasicSingleColumnWithSearch = () => {
 
 BasicSingleColumnWithSearch.storyName = 'basic (single column) with search';
 
+export const BasicSingleColumnWithAsyncLoadingAndSearch = () => {
+  const numberOfItems = number('number of items to render', 100);
+  const generateItems = (number = numberOfItems) =>
+    [...Array(number)].map((_, i) => ({
+      id: `item-${i}`,
+      content: {
+        value: `Item ${i}`,
+        secondaryValue: `Item ${i} Subvalue`,
+      },
+    }));
+
+  const simulateGettingItemsFromServer = (searchValue) =>
+    new Promise((resolve) => {
+      setTimeout(() => {
+        const itemsFromServer = generateItems(numberOfItems).filter(
+          ({ content }) =>
+            content.value.includes(searchValue) || content.secondaryValue.includes(searchValue)
+        );
+        resolve(itemsFromServer);
+      }, 1500);
+    });
+
+  const ListWithAsyncLoadingAndSearch = () => {
+    const [searchValue, setSearchValue] = useState('');
+    const [items, setItems] = useState([]);
+    const [isLoading, setIsLoading] = useState(false);
+
+    useEffect(() => {
+      const searchServerForItems = async () => {
+        setIsLoading(true);
+        const newItems = await simulateGettingItemsFromServer(searchValue);
+        setItems(newItems);
+        setIsLoading(false);
+      };
+
+      searchServerForItems();
+    }, [searchValue]);
+    return (
+      <div style={{ width: 400 }}>
+        <List
+          title={text('title', 'NY Yankees')}
+          items={items}
+          isFiltering={!!searchValue}
+          isLoading={isLoading}
+          search={{
+            onChange: (evt) => setSearchValue(evt.target.value),
+            hasFastSearch: false,
+          }}
+          isVirtualList={boolean('isVirtualList', false)}
+        />
+      </div>
+    );
+  };
+
+  return <ListWithAsyncLoadingAndSearch />;
+};
+
+BasicSingleColumnWithAsyncLoadingAndSearch.storyName = 'with async loading and search';
```

Comment:
nit: I think it is best to keep these names synced so that that the URL and the story name matches e.g.
`WithAsyncLoadingAndSearch.storyName = 'with async loading and search';`

Reply:
works for me! I'll adjust.